[2025-03-21T22:00:08.385+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-03-21T22:00:08.405+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: weather-data-pipeline.check_and_insert_raw_data scheduled__2025-03-21T21:00:00+00:00 [queued]>
[2025-03-21T22:00:08.414+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: weather-data-pipeline.check_and_insert_raw_data scheduled__2025-03-21T21:00:00+00:00 [queued]>
[2025-03-21T22:00:08.415+0000] {taskinstance.py:2867} INFO - Starting attempt 1 of 2
[2025-03-21T22:00:08.428+0000] {taskinstance.py:2890} INFO - Executing <Task(PythonOperator): check_and_insert_raw_data> on 2025-03-21 21:00:00+00:00
[2025-03-21T22:00:08.435+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=3869) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-03-21T22:00:08.437+0000] {standard_task_runner.py:72} INFO - Started process 3871 to run task
[2025-03-21T22:00:08.437+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'weather-data-pipeline', 'check_and_insert_raw_data', 'scheduled__2025-03-21T21:00:00+00:00', '--job-id', '176', '--raw', '--subdir', 'DAGS_FOLDER/weather-data-pipeline.py', '--cfg-path', '/tmp/tmp5jngpg5y']
[2025-03-21T22:00:08.440+0000] {standard_task_runner.py:105} INFO - Job 176: Subtask check_and_insert_raw_data
[2025-03-21T22:00:08.634+0000] {task_command.py:467} INFO - Running <TaskInstance: weather-data-pipeline.check_and_insert_raw_data scheduled__2025-03-21T21:00:00+00:00 [running]> on host c25e9da6226b
[2025-03-21T22:00:08.717+0000] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='weather-data-pipeline' AIRFLOW_CTX_TASK_ID='check_and_insert_raw_data' AIRFLOW_CTX_EXECUTION_DATE='2025-03-21T21:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-03-21T21:00:00+00:00'
[2025-03-21T22:00:08.718+0000] {taskinstance.py:732} INFO - ::endgroup::
[2025-03-21T22:00:08.738+0000] {base.py:84} INFO - Retrieving connection 'rizqi-neon'
[2025-03-21T22:00:10.896+0000] {weather-data-pipeline.py:180} INFO - üîç OpenMeteo Columns: Index(['id', 'datetime', 'temperature', 'area', 'source', 'created_at'], dtype='object')
[2025-03-21T22:00:10.898+0000] {weather-data-pipeline.py:181} INFO - üîç WeatherAPI Columns: Index(['id', 'datetime', 'temperature', 'area', 'source', 'created_at'], dtype='object')
[2025-03-21T22:00:10.898+0000] {weather-data-pipeline.py:182} INFO - üîç Tomorrow.io Columns: Index(['id', 'datetime', 'temperature', 'area', 'source', 'created_at'], dtype='object')
[2025-03-21T22:00:10.899+0000] {weather-data-pipeline.py:184} INFO - 
üîç Data Types:
id                     object
datetime       datetime64[ns]
temperature           float64
area                   object
source                 object
created_at     datetime64[ns]
dtype: object
[2025-03-21T22:00:10.901+0000] {weather-data-pipeline.py:185} INFO - 
üîç Data Types:
id                     object
datetime       datetime64[ns]
temperature           float64
area                   object
source                 object
created_at     datetime64[ns]
dtype: object
[2025-03-21T22:00:10.902+0000] {weather-data-pipeline.py:186} INFO - 
üîç Data Types:
id                     object
datetime       datetime64[ns]
temperature           float64
area                   object
source                 object
created_at     datetime64[ns]
dtype: object
[2025-03-21T22:00:10.903+0000] {weather-data-pipeline.py:188} INFO - 
üìä Shape (Jumlah Data): OpenMeteo: (168, 6), WeatherAPI: (24, 6), Tomorrow.io: (121, 6)
[2025-03-21T22:00:10.908+0000] {weather-data-pipeline.py:196} INFO - 
üöÄ Hasil Akhir (Gabungan DataFrame):
                                 id            datetime  temperature     area      source                 created_at
0  d141d7863df92d8589c38ada109e123d 2025-03-22 00:00:00         26.8  Jakarta  Open Meteo 2025-03-21 22:00:09.698578
1  9a76c33fd475d3dd15980ea9b0a6cf0f 2025-03-22 01:00:00         26.7  Jakarta  Open Meteo 2025-03-21 22:00:09.698578
2  bcecc2bcc96510667af038521c26a926 2025-03-22 02:00:00         26.3  Jakarta  Open Meteo 2025-03-21 22:00:09.698578
3  bfbcf0e41660e298c99997d3d53d7d1c 2025-03-22 03:00:00         25.9  Jakarta  Open Meteo 2025-03-21 22:00:09.698578
4  b50fabebec04f1c1ceae4f21a3c1fe62 2025-03-22 04:00:00         25.7  Jakarta  Open Meteo 2025-03-21 22:00:09.698578
[2025-03-21T22:00:11.104+0000] {python.py:240} INFO - Done. Returned value was: None
[2025-03-21T22:00:11.156+0000] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-03-21T22:00:11.158+0000] {taskinstance.py:353} INFO - Marking task as SUCCESS. dag_id=weather-data-pipeline, task_id=check_and_insert_raw_data, run_id=scheduled__2025-03-21T21:00:00+00:00, execution_date=20250321T210000, start_date=20250321T220008, end_date=20250321T220011
[2025-03-21T22:00:11.225+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-03-21T22:00:11.476+0000] {taskinstance.py:3901} INFO - 2 downstream tasks scheduled from follow-on schedule check
[2025-03-21T22:00:11.489+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
